pkg/utils/options/options.go

```diff
@@ -39,6 +39,8 @@ func MustParse() Options {
 	flag.IntVar(&opts.WebhookPort, "port", 8443, "The port the webhook endpoint binds to for validation and mutation of resources")
 	flag.IntVar(&opts.KubeClientQPS, "kube-client-qps", env.WithDefaultInt("KUBE_CLIENT_QPS", 200), "The smoothed rate of qps to kube-apiserver")
 	flag.IntVar(&opts.KubeClientBurst, "kube-client-burst", env.WithDefaultInt("KUBE_CLIENT_BURST", 300), "The maximum allowed burst of queries to the kube-apiserver")
+	flag.StringVar(&opts.IpFamily, "ip-family", env.WithDefaultString("IP_FAMILY", ""), "The kubernetes ip family (ipv4/ipv6)")
+	flag.StringVar(&opts.ServiceIpv6Cidr, "service-ipv6-cidr", env.WithDefaultString("SERVICE_IPV6_CIDR", ""), "The kubernetes service CIDR for IPv6 cluster")
```

Comment:
I wonder if we can do something like https://pkg.go.dev/net#ParseCIDR to reduce our API Surface. I could imagine that this configuration lives in our provisioner API itself, rather than the helm chart: https://github.com/aws/karpenter/blob/main/pkg/apis/provisioning/v1alpha5/kubelet_configuration.go

Reply:
I agree that we can resolve IP Family from a service cidr:

https://github.com/aws/karpenter/blob/f1f9e51fd024fd138df32b87ccb1199051331a35/pkg/cloudprovider/aws/launchtemplate.go#L262-L277